src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -58,9 +65,30 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         auto clickedItemContainer = args.InvokedItemContainer();
 
-        if (clickedItemContainer != NULL)
+        if (clickedItemContainer)
         {
-            Navigate(contentFrame(), unbox_value<hstring>(clickedItemContainer.Tag()));
+            auto tagPropertyValue = clickedItemContainer.Tag().as<winrt::Windows::Foundation::IPropertyValue>();
+            if (tagPropertyValue.Type() == winrt::Windows::Foundation::PropertyType::String)
+            {
+                auto tag = unbox_value<hstring>(clickedItemContainer.Tag());
+                if (tag == L"AddProfile")
+                {
+                    uint32_t insertIndex;
+                    SettingsNav().MenuItems().IndexOf(clickedItemContainer, insertIndex);
+                    _CreateAndNavigateToNewProfile(insertIndex);
+                }
+                else
+                {
+                    Navigate(contentFrame(), unbox_value<hstring>(clickedItemContainer.Tag()));
+                }
+            }
+            else if (tagPropertyValue.Type() == winrt::Windows::Foundation::PropertyType::Guid)
+            {
+                auto profileGuid = unbox_value<guid>(clickedItemContainer.Tag());
+                auto prof = _settingsSource.FindProfile(profileGuid);

```

Comment:
Users can technically change the profile guid, right? Would we want to create a map of `Profile`--> `NavViewItem` instead of `guid`-->`NavViewItem`?

Reply:
oh good catch, yeah guid isn't the best indicator, so instead of stuffing guid in the Tag, I stuffed the Profile object itself, lmk if I'm crazy though